worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -1514,6 +1515,25 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
       hardSplitIndexes: Array[Int] = Array.empty[Int]): Unit = {
     val length = fileWriters.length
     val result = new Array[StatusCode](length)
+
+    var finalBody: ByteBuf = body
```

Comment:
Copying and quickly releasing buffers can indeed reduce direct memory usage. However, I prefer to perform this in the PartitionWriter before generating the FlushTask, because after the FlushTask is completed, the chunk memory for that part can be completely released. With the current PR, the memory cannot be released until all related tasks are finished. In addition, I recommend copying the buffer only after the direct memory exceeds a certain threshold, so that when the system is not under heavy pressure, we can reduce unnecessary CPU consumption.